src/ResourceManager/Sql/Commands.Sql/VirtualNetworkRule/Cmdlet/NewAzureSqlServerVirtualNetworkRule.cs

```diff
@@ -0,0 +1,105 @@
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmSqlServerVirtualNetworkRule",
+        ConfirmImpact = ConfirmImpact.Low, SupportsShouldProcess = true)]
+    public class NewAzureSqlServerVirtualNetworkRule : AzureSqlServerVirtualNetworkRuleCmdletBase
+    {
+        /// <summary>
+        /// Azure Sql Server Virtual Network Rule Name.
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "Azure Sql Server Virtual Network Rule Name.")]
+        [ValidateNotNullOrEmpty]
+        public string VirtualNetworkRuleName { get; set; }
+
+        /// <summary>
+        /// The Virtual Network Subnet Id
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The Virtual Network Subnet Id that specifies the Microsoft.Network details")]
+        [ValidateNotNull]
+        public string VirtualNetworkSubnetId { get; set; }
+
+        /// <summary>
+        /// Check to see if the Virtual Network Rule already exists for this server
+        /// </summary>
+        /// <returns>Null if the Virtual Network Rule doesn't exist.  Otherwise throws exception</returns>
+        protected override IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> GetEntity()
+        {
+            try
+            {
+                ModelAdapter.GetVirtualNetworkRule(this.ResourceGroupName, this.ServerName, this.VirtualNetworkRuleName);
+            }
+            catch (CloudException ex)
+            {
+                if (ex.Response.StatusCode == System.Net.HttpStatusCode.NotFound)
+                {
+                    // This is what we want.  We looked and there is no server with this name.
+                    return null;
+                }
+
+                // Unexpected exception encountered
+                throw;
+            }
+
+            // The server already exists
+            throw new PSArgumentException(
+                string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.ServerVirtualNetworkRuleNameExists, this.VirtualNetworkRuleName, this.ServerName),
+                "VirtualNetworkRule");
+        }
+
+        /// <summary>
+        /// Generates the model from user input.
+        /// </summary>
+        /// <param name="model">This is null since the server doesn't exist yet</param>
+        /// <returns>The generated model from user input</returns>
+        protected override IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> ApplyUserInputToModel(IEnumerable<Model.AzureSqlServerVirtualNetworkRuleModel> model)
+        {
+            List<Model.AzureSqlServerVirtualNetworkRuleModel> newEntity = new List<Model.AzureSqlServerVirtualNetworkRuleModel>();
+            newEntity.Add(new Model.AzureSqlServerVirtualNetworkRuleModel()
+            {
+                ResourceGroupName = this.ResourceGroupName.Trim(),
+                ServerName = this.ServerName.Trim(),
+                VirtualNetworkRuleName = this.VirtualNetworkRuleName.Trim(),
```

Comment:
is it standard to trim like this? I wonder if somebody does `new-vnetrule -rulename "asdf   "` and then `'get-vnetrule -rulename "asdf  "` will they get the rule back? I mean I think Trim has to be done everywhere or nowhere.

Reply:
According to my tests it should work. But you are right, ideally it would be better to have Trim everywhere. I will make that change.